plugins/modeltraining/pkg/modeltraining/plugin.go

```diff
@@ -0,0 +1,95 @@
+package modeltraining
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	opensearch "github.com/opensearch-project/opensearch-go"
+	managementv1 "github.com/rancher/opni/pkg/apis/management/v1"
+	"github.com/rancher/opni/pkg/logger"
+	managementext "github.com/rancher/opni/pkg/plugins/apis/apiextensions/management"
+	"github.com/rancher/opni/pkg/plugins/apis/system"
+	"github.com/rancher/opni/pkg/plugins/meta"
+	"github.com/rancher/opni/pkg/util"
+	"github.com/rancher/opni/pkg/util/future"
+	"github.com/rancher/opni/plugins/modeltraining/pkg/apis/modeltraining"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ModelTrainingPlugin struct {
+	modeltraining.UnsafeModelTrainingServer
+	system.UnimplementedSystemPluginClient
+	ctx            context.Context
+	Logger         *zap.SugaredLogger
+	k8sClient      client.Client
+	osClient       future.Future[*opensearch.Client]
+	natsConnection future.Future[*nats.Conn]
```

Comment:
You don't really need to store the natsConnection in state, just the KV as the connection is only used to create that KV

Reply:
Actually, the Nats connection is used in the api.go function to send over payload to the AI training controller service. 
Example: https://github.com/rancher/opni/blob/training-gateway-plugin/plugins/modeltraining/pkg/modeltraining/api.go#L42